src/orange/detail/BIHBuilder.hh

```diff
@@ -53,36 +54,42 @@ class BIHBuilder
     explicit BIHBuilder(Storage* storage);
 
     // Create BIH Nodes
-    BIHTree operator()(VecBBox bboxes);
+    BIHTree operator()(VecBBox&& bboxes);
 
   private:
     /// TYPES ///
 
     using Real3 = Array<fast_real_type, 3>;
-    using VecReal3 = std::vector<Real3>;
     using VecIndices = std::vector<LocalVolumeId>;
-    using PairVecIndices = std::pair<VecIndices, VecIndices>;
-    using AxesCenters = std::vector<std::vector<real_type>>;
     using VecNodes = std::vector<std::variant<BIHInnerNode, BIHLeafNode>>;
     using VecInnerNodes = std::vector<BIHInnerNode>;
     using VecLeafNodes = std::vector<BIHLeafNode>;
     using ArrangedNodes = std::pair<VecInnerNodes, VecLeafNodes>;
 
+    struct Temporaries
+    {
+        VecBBox bboxes;
+        std::vector<Real3> centers;
+    };
+
     //// DATA ////
 
-    VecBBox bboxes_;
-    VecReal3 centers_;
-    Storage* storage_;
+    Temporaries temp_;
+
+    CollectionBuilder<FastBBox> bboxes_;
+    CollectionBuilder<LocalVolumeId> local_volume_ids_;
+    CollectionBuilder<BIHInnerNode> inner_nodes_;
+    CollectionBuilder<BIHLeafNode> leaf_nodes_;
```

Comment:
To be more consistent with the other builders, I've made `foo_` a builder that targets `data.foo`. To avoid the conflict with the temporary class data bboxes/centers I put those in a custom class.

Later we can turn these `CollectionBuilders` to `DedupeCollectionBuilders`.